data/moves.ts

```diff
@@ -991,7 +991,9 @@ export const BattleMovedex: {[moveid: string]: MoveData} = {
 			onTryHitPriority: 3,
 			onTryHit(target, source, move) {
 				if (!move.flags['protect']) {
-					if (move.isZ || move.isMax) target.getMoveHitData(move).zBrokeProtect = true;
+					if (move.isZ || (move.isMax && !['gmaxoneblow', 'gmaxrapidflow'].includes(move.id))) {
+						target.getMoveHitData(move).zBrokeProtect = true;
```

Comment:
The description for those moves says that they break protection. Assuming that's accurate, this suggests that the protection should be removed before the move hits, and then this change (and similar occurrences) won't be necessary.

Reply:
I believe i did it correctly, do you think I should move the move check inside of the `if` statement but return before `zBrokeProtect`?

Reply:
I'd better wait until I know the exact mechanics.

Reply:
I've changed my mind on this.

I now think the protection breaking code should reset the broke through protect flag.